### PR TITLE
Fix: Hours can have a "or" in them.

### LINKED ISSUE #119: Hours can have a "or" in them
Need to be able to handle "or" in hours. 
https://clockwork.vikelabs.dev/calendar/202101/PHIL?pid=HkxKERd67E

https://www.uvic.ca/calendar/archives/202101/undergrad/index.php#/courses/HkxKERd67E
Currently the resulting output will become `3-0-0 or 2`.

### PR DESCRIPTION
**Pull request by NishchintDhawan**
Fix for the issue #119: Hours can have a "or" in them.

### CODE REVIEW

File: src/kuali/__tests__/catalog.test.ts [hunk 1]
```diff
@@ -14,6 +14,24 @@ describe('KualiCatalogItemParser', () => {
...
+        //   console.log(values[0]);
+        // }
+        
+        expect(response.hoursCatalog).toStrictEqual([{ lecture: '3', lab: '0', tutorial: '0' },{ lecture: '2', lab: '1', tutorial: '0' }]);
+        expect(response.credits.credits).toStrictEqual({ min: '1.5', max: '1.5' });
+        expect(response.credits.value).toEqual('1.5');
+        expect(response.credits.chosen).toEqual('fixed');
+      });
+    })});
+
```
Comment (aomi) [inline src/kuali/__tests__/catalog.test.ts:47]: Adding a test case for the single case (so only one hour catalog) would be good just to be safe

File: src/kuali/catalog.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import * as cheerio from 'cheerio';
 
 import { Course, KualiCourseItem, KualiCourseItemParsed, NestedPreCoRequisites } from '../types';
+import { each } from 'async';
```
Comment (aomi) [inline src/kuali/catalog.ts]: You aren't using this so we should remove it.

File: src/kuali/catalog.ts [hunk 3]
```diff
@@ -92,16 +92,32 @@ function parsePreCoReqs(preCoReqs: string): Array<NestedPreCoRequisites | Course
   return reqs;
 }
 
+function set_hoursCatalog(hours: string[]){
```
Comment (aomi) [inline src/kuali/catalog.ts]: I would argue this isn't "setting" anything and doesn't reflect what this function does.

**Review by aomi — COMMENTED** [review]
Thanks for the contribution. Some general notes.
- You have to add spaces between the linked issue in the description or it won't link properly.
- Avoid using `var` given it isn't required and we try to use `const` and `let` wherever we can.
- Please remove any commented code prior to submitting your PR. 
- Please use `camelCase` instead of `under_score` for function names and variables.

**Review by keithradford — COMMENTED** [review]
Code looks good, but there's some general formatting that needs to be addressed to keep things consistent with the rest of the codebase. Running `npm run lint:fix` should address most if not all of them

File: src/kuali/catalog.ts [hunk 4]
```diff
@@ -91,32 +92,32 @@ function parsePreCoReqs(preCoReqs: string): Array<NestedPreCoRequisites | Course
   return reqs;
 }
 
-function hoursCatalog(hours: string[]){
-  let eachHour: {lecture: string, lab: string, tutorial: string}[];
-  eachHour=[];
-  
+function hoursCatalog(hours: string[]) {
+  let eachHour: { lecture: string; lab: string; tutorial: string }[];
+  eachHour = [];
```
Comment (keithradford) [inline src/kuali/catalog.ts]:
```suggestion
  const eachHour: { lecture: string; lab: string; tutorial: string }[] = [];
```

File: src/kuali/catalog.ts [hunk 5]
```diff
@@ -91,32 +92,32 @@ function parsePreCoReqs(preCoReqs: string): Array<NestedPreCoRequisites | Course
...
+
   //store the hours in a new array.
-  hours.forEach(element => {
-    let temp=element.split('-');
-    let JSONvalue = {lecture: temp[0], lab: temp[1], tutorial: temp[2]};
-    eachHour= eachHour.concat(JSONvalue);
+  hours.forEach((element) => {
+    const temp = element.split('-');
+    const JSONvalue = { lecture: temp[0], lab: temp[1], tutorial: temp[2] };
+    eachHour = eachHour.concat(JSONvalue);
```
Comment (keithradford) [inline src/kuali/catalog.ts]:
```suggestion
    eachHour.push({ lecture: temp[0], lab: temp[1], tutorial: temp[2] });
```

**Review by keithradford — COMMENTED** [review]
Just a few minor things

**Review by keithradford — APPROVED** [review]
LGTM!

**Pull request merged**